### PR TITLE
Handle footnotes in API docs

### PR DESCRIPTION
**Pull request by Eric-Arellano**
Closes https://github.com/Qiskit/documentation/issues/1978.

### CODE REVIEW

File: docs/api/qiskit/release-notes/1.1.mdx [hunk 1]
```diff
@@ -910,7 +910,11 @@ The Qiskit 1.1.0 release is a minor feature release that includes a myriad of ne
 
 ### Other Notes
 
-*   Support for the arm64 macOS platform has been promoted from Tier 3 to Tier 1. Previously the platform was at Tier 3 because there was no available CI environment for testing Qiskit on the platform. Now that Github has made an arm64 macOS environment available to open source projects [\[1\]](#id9) we’re testing the platform along with the other Tier 1 supported platforms.
+*   Support for the arm64 macOS platform has been promoted from Tier 3 to Tier 1. Previously the platform was at Tier 3 because there was no available CI environment for testing Qiskit on the platform. Now that Github has made an arm64 macOS environment available to open source projects <span id="id8" />[\[1\]](#id9) we’re testing the platform along with the other Tier 1 supported platforms.
+
+    ## Footnotes
```
Comment (Eric-Arellano) [inline docs/api/qiskit/release-notes/1.1.mdx:915]: Hm, @frankharkins this wasn't expected in a release note. What do you think? Should we skip adding the h2 if it's a release note file?

File: docs/api/qiskit/0.46/qiskit.algorithms.optimizers.NFT.mdx [hunk 2]
```diff
@@ -28,10 +28,14 @@ python_api_name: qiskit.algorithms.optimizers.NFT
 
   **Notes**
 
-  In this optimization method, the optimization function have to satisfy three conditions written in [\[1\]](#id2).
+  In this optimization method, the optimization function have to satisfy three conditions written in   <span id="id1" />[\[1\]](#id2).
 
   **References**
 
+  ## Footnotes
```
Comment (Eric-Arellano) [inline docs/api/qiskit/0.46/qiskit.algorithms.optimizers.NFT.mdx:35]: We probably shouldn't be adding the h2 here: there is already the references description right above, and h2 messes up the information hierarchy.

**Comment by Eric-Arellano** [comment]
@frankharkins after reviewing the two edge cases I commented on above, I'm thinking our code should not add the h2. We can't anticipate correctly what the surrounding context is for where the footnotes get listed.

Instead, we should update qpy's RST to set an h2 directly in this file:

https://github.com/Qiskit/qiskit/blob/ee2bd5c9ba834d40c1097e8dc118049268dfc9a0/qiskit/qpy/__init__.py#L1328-L1333

Wdyt?

**Review by frankharkins — APPROVED** [review]
Thank you!

**Pull request merged**